ui/components/DashboardComponent.js

```diff
@@ -137,25 +140,26 @@ class DashboardComponent extends React.Component {
 
   static getDerivedStateFromProps(props, state) {
     const {
-      meshAdapters, meshAdaptersts, k8sconfig, grafana, prometheus
+      meshAdapters, meshAdaptersts, k8sconfig, grafana, prometheus, contextName
     } = props;
     const st = {};
     if (meshAdaptersts > state.mts) {
       st.meshAdapters = meshAdapters;
       st.mts = meshAdaptersts;
     }
+    st.grafana = grafana;
+    st.prometheus = prometheus;
     if (k8sconfig.ts > state.kts) {
       st.inClusterConfig = k8sconfig.inClusterConfig;
       st.k8sfile = k8sconfig.k8sfile;
       st.contextName = k8sconfig.contextName;
       st.clusterConfigured = k8sconfig.clusterConfigured;
       st.configuredServer = k8sconfig.configuredServer;
       st.kts = props.ts;
+      if (!state.contextsFromFile?.length)
+        st.contextsFromFile = { ...st, contextsFromFile : [{ contextName, currentContext : true }] };
```

Comment:
Is this what you wanted to do?
```suggestion
        st.contextsFromFile = { ...(st.contextsFromFile || []), contextsFromFile : [{ contextName, currentContext : true }] };
```

Reply:
:thinking:, ..yes